javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,367 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class LinkedList[E]() extends AbstractSequentialList[E]
+    with List[E] with Deque[E] with Cloneable with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  import LinkedList._
+
+  private var head: Node[E] = null
+  private var last: Node[E] = null
+
+  private var _size = 0
+
+  def getFirst(): E =
+    if (_size <= 0)
+      throw new NoSuchElementException()
+    else peekFirst()
+
+  def getLast(): E =
+    if (_size <= 0)
+      throw new NoSuchElementException()
+    else peekLast()
+
+  def removeFirst(): E = {
+    if (isEmpty())
+      throw new NoSuchElementException()
+
+    val oldHead = head
+    head = oldHead.next
+
+    if (head ne null)
+      head.prev = null
+    else
+      last = null
+
+    _size -= 1
+    oldHead.value
+  }
+
+  def removeLast(): E = {
+    if (isEmpty())
+      throw new NoSuchElementException()
+
+    val oldLast = last
+    last = oldLast.prev
+
+    if (last ne null)
+      last.next = null
+    else
+      head = null
+
+    _size -= 1
+    oldLast.value
+  }
+
+  def addFirst(e: E): Unit = {
+    val oldHead = head
+
+    head = new Node(e, next = oldHead)
+
+    _size += 1
+
+    if (oldHead ne null)
+      oldHead.prev = head
+    else
+      last = head
+  }
+
+  def addLast(e: E): Unit = {
+    val oldLast = last
+
+    last = new Node(e, prev = oldLast)
+
+    _size += 1
+
+    if (oldLast ne null)
+      oldLast.next = last
+    else
+      head = last
+  }
+
+  override def contains(o: Any): Boolean =
+    iterator().exists(_ === o)
+
+  override def size(): Int =
+    _size
+
+  override def add(e: E): Boolean = {
+    addLast(e)
+    true
+  }
+
+  override def remove(o: Any): Boolean =
+    _removeOccurrence(listIterator, o)
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator
+    val changed = iter.hasNext()
+    while (iter.hasNext())
+      addLast(iter.next())
+
+    changed
+  }
+
+  override def clear(): Unit = {
+    head = null
+    last = null
+    _size = 0
+  }
+
+  private def getNodeAt(index: Int): Node[E] = {
+    if (index == 0) head
+    else if (index == size - 1) last
+    else {
+      var current: Node[E] = null
+      if (index <= size/2) {
+        current = head
+        for (_ <- 0 until index) current = current.next
+      } else {
+        current = last
+        for (_ <- index until (size - 1)) current = current.prev
```

Comment:
The bodies of the `for` loops should be on their dedicated lines, as shown in https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#for-comprehensions